tools/amghrcvr/main.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/google/go-github/v26/github"
+	"github.com/prometheus/alertmanager/notify"
+	"golang.org/x/oauth2"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var (
+	authfile       string
+	defaultOwner   string
+	defaultRepo    string
+	ghClient       *github.Client
+	issueSvc       *github.IssuesService
+	ctx, cancelCtx = context.WithCancel(context.Background())
+)
+
+func processAlert(msg *notify.WebhookMessage) error {
+
+	// add comment if firing
+	if msg.Data.Status == "firing" {
+		msgBody, err := formatIssueBody(msg)
+		if err != nil {
+			return err
+		}
+
+		prNo, err := getTargetPR(msg)
+		if err != nil {
+			return err
+		}
+		issueComment := github.IssueComment{Body: &msgBody}
+		_, _, err = issueSvc.CreateComment(ctx, getTargetOwner(msg), getTargetRepo(msg), prNo, &issueComment)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func handleWebhook(rw http.ResponseWriter, r *http.Request) {
+
+	if r.Method != http.MethodPost {
+		log.Printf("Client used unsupported method: %s: %s", r.Method, r.RemoteAddr)
```

Comment:
I usually use %v so that the log doesn't brake if you change the input by mistake.
```suggestion
		log.Printf("unsupported request method: %v: %v", r.Method, r.RemoteAddr)
```